backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -143,9 +155,9 @@ private Folder getFolder(Session session, String commitSha, String folderSha) {
         Folder folder = getFolder(session, commit.getCommit_hash(), folderHash);
         session.getTransaction().commit();
         if (folder == null) { // root is empty
-          return GetCommitComponentRequest.Response.newBuilder().build();
+          return new AbstractMap.SimpleEntry<>(Response.newBuilder().build(), null);
         }
-        return GetCommitComponentRequest.Response.newBuilder().setFolder(folder).build();
+        return new AbstractMap.SimpleEntry<>(Response.newBuilder().setFolder(folder).build(), null);
```

Comment:
What is this achieving? Seems risky to send null.

Reply:
I just edited existing code to return blob information. I agree that it risky, I have added them. You meant that I should use optionals there? I have added them now.